hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/SimpleRegionNormalizer.java

```diff
@@ -540,24 +546,34 @@ public boolean isMergeEnabled() {
       return mergeEnabled;
     }
 
-    public int getMinRegionCount() {
-      return minRegionCount;
+    public int getMergeMinRegionCount() {
+      return mergeMinRegionCount;
     }
 
-    public int getMinRegionCount(NormalizeContext context) {
-      int minRegionCount = context.getOrDefault(MIN_REGION_COUNT_KEY, Integer::parseInt, 0);
-      if (minRegionCount <= 0) {
-        minRegionCount = getMinRegionCount();
+    public int getMergeMinRegionCount(NormalizeContext context) {
+      String stringValue = context.getOrDefault(MERGE_MIN_REGION_COUNT_KEY,
+        Function.identity(), null);
+      if (stringValue == null) {
+        stringValue = context.getOrDefault(MIN_REGION_COUNT_KEY, Function.identity(),  null);
+        if (stringValue != null) {
+          LOG.debug("The config key {} in table descriptor is deprecated. Instead please use {}. "
```

Comment:
huh. maybe `TableDescriptor` should handle deprecations as well. For another patch.